components/AddDonation.js

```diff
@@ -0,0 +1,136 @@
+import styled from "styled-components";
+import { useState } from "react";
+import Counter from "./Counter";
+
+export default function AddDonation({ createDonation }) {
+  const [toggleAddButton, setToggleAddButton] = useState(true);
+  const [count, setCount] = useState(0);
+  function handleClick() {
+    setToggleAddButton(!toggleAddButton);
+  }
+  function handleSubmit(event) {
+    event.preventDefault();
+
+    const formData = new FormData(event.target);
+    const data = Object.fromEntries(formData);
+
+    createDonation(data);
+    setCount(0);
+    event.target.reset();
+    setToggleAddButton(!toggleAddButton);
+  }
+
+  return (
+    <StyledDiv>
+      {toggleAddButton ? (
+        <StyledButton onClick={handleClick}>
+          Neue Spende hinzufügen
+        </StyledButton>
+      ) : (
+        <StyledForm onSubmit={handleSubmit}>
+          <label htmlFor="category">Kategorie:</label>
+          <select name="category" id="category" required>
+            <option value="">-Wähle bitte eine Kategorie-</option>
+            <option value="fruits&vegetables">Obst & Gemüse</option>
+            <option value="homemade">Selbstgemacht</option>
+            <option value="pastry">Backwaren</option>
+            <option value="dairy">Käse, Eier & Molkerei</option>
+            <option value="frozen">Tiefkühlkost</option>
+            <option value="sweet&salty">Süßes & Salziges</option>
+            <option value="beverage">Getränke</option>
+            <option value="meat&fish">Fleisch & Fisch</option>
```

Comment:
at some point you might generate these options from an optionsOIbject. This is useful if you want to dynamically add options. But for now its ok.